src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -82,13 +82,35 @@ public static function fromDsn(string $dsn, array $options = array(), bool $debu
         }
 
         $exchangeOptions = $amqpOptions['exchange'];
-        $queueOptions = $amqpOptions['queue'];
+        $queueOptions = self::getQueueOptions($amqpOptions['queue']);
 
         unset($amqpOptions['queue'], $amqpOptions['exchange']);
 
         return new self($amqpOptions, $exchangeOptions, $queueOptions, $debug, $amqpFactory);
     }
 
+    private static function getQueueOptions(array $queueOptions): array
+    {
+        if (!array_key_exists('arguments', $queueOptions)) {
+            return $queueOptions;
+        }
+
+        foreach ($queueOptions['arguments'] as $key => $value) {
+            switch ($key) {
+                case 'x-message-ttl':
+                case 'x-max-length':
+                case 'x-max-length-bytes':
+                case 'x-max-priority':
+                    // RabbitMQ expects the arguments above to be passed as integer values
+                    $queueOptions['arguments'][$key] = (int) $value;
+
+                    break;
```

Comment:
Shouldn't there be a `default` that would throw?

Reply:
@greg0ire  Why should the `default` throw an exception? Other parameters (not listed here) should be used verbatim

Reply:
Oh ok nevermind then